Xamarin.Forms.Platform.Android/FastRenderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,488 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+
+using Android.App;
+using Android.Content;
+using Android.OS;
+using Android.Runtime;
+using Android.Views;
+using AView = Android.Views.View;
+using Android.Widget;
+using Xamarin.Forms.Internals;
+using Android.Media;
+
+namespace Xamarin.Forms.Platform.Android.FastRenderers
+{
+	internal sealed class MediaElementRenderer : FrameLayout, IVisualElementRenderer, IViewRenderer, IEffectControlProvider, MediaPlayer.IOnCompletionListener, MediaPlayer.IOnInfoListener, MediaPlayer.IOnPreparedListener, MediaPlayer.IOnErrorListener
+	{
+		bool _isDisposed;
+		int? _defaultLabelFor;
+		MediaElement MediaElement { get; set; }
+		IMediaElementController Controller => MediaElement as IMediaElementController;
+
+		readonly AutomationPropertiesProvider _automationPropertiesProvider;
+		readonly EffectControlProvider _effectControlProvider;
+		VisualElementTracker _tracker;
+
+		MediaController _controller;
+		FormsVideoView _view;
+		MediaPlayer _mediaPlayer;
+
+		public MediaElementRenderer(Context context) : base(context)
+		{
+			_automationPropertiesProvider = new AutomationPropertiesProvider(this);
+			_effectControlProvider = new EffectControlProvider(this);
+
+			Initialize();
+		}
+
+		public VisualElement Element => MediaElement;
+
+		VisualElementTracker IVisualElementRenderer.Tracker => _tracker;
+
+		ViewGroup IVisualElementRenderer.ViewGroup => null;
+
+		AView IVisualElementRenderer.View => this;
+
+		public event EventHandler<VisualElementChangedEventArgs> ElementChanged;
```

Comment:
Can we order public stuff then protected, then private.. helps reading this code.